### PR TITLE
Removing PolicyGroup, simplifying Policy.  Introducing Contract.  Removing ChallengePack.

### PR DESCRIPTION
**Pull request by jMyles**

### CODE REVIEW

File: nkms/policy/models.py [hunk 1]
```diff
@@ -11,146 +13,51 @@
...
-
-class PolicyOffer(object):
+class Contract(object):
     """
-    An offer from Alice to Ursula to enter into a contract for Re-Encryption services.
+    A Policy must be implemented by agreement with n Ursulas.  This class tracks the status of that implementation.
     """
 
-    def __init__(self, n, deposit, contract_end_datetime):
+    def __init__(self, alice, hrac, expiration, deposit=None, ursula=None, kfrag=UNKNOWN_KFRAG, alices_signature=None):
```
[the remark this answers is not in the document]
Comment (jMyles) [reply nkms/policy/models.py]: The flow here is like this:

* Alice makes a Contract without a KFrag
* Alice sends the Contract to Ursula to inquire whether Ursula is agreeable
* Ursula sends back a signed response, indicating that she's game
* Alice repeats this until she has N agreeable Ursulas
* Alice attached KFrags to the N Contracts corresponding to those N Ursulas
* Alice signs the Contract, with KFrag attached, and sends it to Ursula

[the remark this answers is not in the document]
Comment (tuxxy) [reply nkms/policy/models.py]: Ah, I see! Cool!

File: nkms/policy/models.py [hunk 2]
```diff
@@ -165,35 +72,33 @@ class Policy(object):
...
         self.alices_signature = alices_signature
-        self.kfrag = kfrag
-        self.random_id_portion = api.secure_random(32)  # TOOD: Where do we actually want this to live?
-        self.challenge_size = challenge_size
-        self.treasure_map = []
-        self.challenge_pack = []
 
-        self._encrypted_challenge_pack = encrypted_challenge_pack
+
+    class MoreContractsThanKFrags(TypeError):
```
[the remark this answers is not in the document]
Comment (jMyles) [reply nkms/policy/models.py]: I actually prefer moving them around with classes instead - ask yourself which is more readable among these two:

`except exceptions.MoreContractsThanKFrags`

or 

`except Policy.MoreContractsThanKFrags`

File: nkms/policy/models.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (jMyles) [reply nkms/policy/models.py]: michael-ahh-i-see-tm

File: nkms/policy/models.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (tuxxy) [reply nkms/policy/models.py]: Ah, I get your point. I think they're both equally readable given that the error condition is provided in the code. I just prefer to have them in their own explicit file. :p

**Pull request merged**